src/main/java/org/wildfly/security/auth/realm/token/TokenSecurityRealm.java

```diff
@@ -124,7 +124,17 @@ public boolean verifyEvidence(Evidence evidence) throws RealmUnavailableExceptio
                 return false;
             }
 
-            return strategy.validate((BearerTokenEvidence) evidence) != null;
+            BearerTokenEvidence tokenEvidence = BearerTokenEvidence.class.cast(evidence);
+
+            try {
+                return strategy.validate(tokenEvidence) != null;
+            } catch (RealmUnavailableException rue) {
+                throw rue;
+            } catch (Exception unknown) {
```

Comment:
It's better to catch each expected exception type here I think.  Or, maybe move the exception catching logic to ```strategy.validate()``` to keep this code simple.